packages/forms/src/Components/ModalTableSelect.php

```diff
@@ -301,9 +300,8 @@ public function relationship(string | Closure | null $name = null, string | Clos
                 $relatedRecords = $relationship->getResults();
 
                 $component->state(
-                    // @js() in the frontend casts numbers to strings.
-                    // Therefore, we need to cast array values to strings, too.
-                    // Otherwise, selection and deselection will break.
+                    // Cast the related keys to a string, otherwise JavaScript does not
+                    // know how to handle deselection.
```

Comment:
@danharrin I hoped to improve this comment a bit, adding the root cause of it (`@js`). Also this affects selection AND deselection. 

I was a bit confused by "otherwise JavaScript does not know how to handle deselection" because it doesn't describe the why.